packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -134,6 +134,7 @@ export class ExecuteTxnRpc extends AccountRpcController<
         callback: async (contractAddress: string, transactionHash: string) => {
           await this.updateAccountAsDeploy(contractAddress, transactionHash);
         },
+        version: details?.version as unknown as constants.TRANSACTION_VERSION,
```

Comment:
it looks like we need an define in superstruct on txVersionStruct to convert it to TRANSACTION_VERSION, hence we dont need to cast it every where

Reply:
the cast comes from the UniversalDetailsStruct that is defined based on starknet UniversalDetails. But there the version is not `constants.TRANSACTION_VERSION` but simply `BigNumberish`, hence the cast.

Reply:
ah i see, then may be but it in a single var in this file, as there are more than 1 place using the casting in the same file?